experiments/LES/dycoms.jl

```diff
@@ -0,0 +1,191 @@
+    P_sfc      = FT(MSLP)
+
+    # Specify moisture profiles
+    q_liq      = FT(0)
+    q_ice      = FT(0)
+    zb         = FT(600)         # initial cloud bottom
+    zi         = FT(840)         # initial cloud top
+
+    if z <= zi
+        θ_liq  = FT(289)
+        q_tot  = qref
+    else
+        θ_liq  = FT(297.5) + (z - zi)^(FT(1/3))
+        q_tot  = FT(1.5e-3)
+    end
+
+    ugeo = FT(7)
+    vgeo = FT(-5.5)
+    u, v, w = ugeo, vgeo, FT(0)
+
+    # Perturb initial state to break symmetry and trigger turbulent convection
+    r1 = FT(rand(Uniform(-0.02, 0.02)))
+    r2 = FT(rand(Uniform(-0.000015, 0.000015)))
+    r3 = FT(rand(Uniform(-0.1, 0.1)))
+    r4 = FT(rand(Uniform(-0.1, 0.1)))
+    if z <= 400.0
+        θ_liq += r1 * θ_liq
+        q_tot += r2 * q_tot
+        u     += r3 * u
+        v     += r4 * v
+    end
+
+    # Pressure
+    H     = Rm_sfc * T_sfc / grav
+    p     = P_sfc * exp(-z / H)
+
+    # Density, Temperature
+    ts    = LiquidIcePotTempSHumEquil_given_pressure(θ_liq, p, q_tot)
+    ρ     = air_density(ts)
+
+    e_kin = FT(1/2) * FT((u^2 + v^2 + w^2))
+    e_pot = grav * z
+    E     = ρ * total_energy(e_kin, e_pot, ts)
+
+    state.ρ               = ρ
+    state.ρu              = SVector(ρ*u, ρ*v, ρ*w)
+    state.ρe              = E
+    state.moisture.ρq_tot = ρ * q_tot
+
+    return nothing
+end
+
+function config_dycoms(FT, N, resolution, xmax, ymax, zmax)
+    # Reference state
+    T_min   = FT(289)
+    T_s     = FT(290.4)
+    Γ_lapse = FT(grav/cp_d)
+    T       = LinearTemperatureProfile(T_min, T_s, Γ_lapse)
+    rel_hum = FT(0)
+    ref_state = HydrostaticState(T, rel_hum)
+
+    # Radiation model
+    κ             = FT(85)
+    α_z           = FT(1)
+    z_i           = FT(840)
+    ρ_i           = FT(1.13)
+    D_subsidence  = FT(3.75e-6)
+    F_0           = FT(70)
+    F_1           = FT(22)
+    radiation = DYCOMSRadiation{FT}(κ, α_z, z_i, ρ_i, D_subsidence, F_0, F_1)
+
+    # Sources
+    f_coriolis    = FT(1.03e-4)
+    u_geostrophic = FT(7.0)
+    v_geostrophic = FT(-5.5)
+    w_ref         = FT(0)
+    u_relaxation  = SVector(u_geostrophic, v_geostrophic, w_ref)
+    # Sponge
+    c_sponge = 1
+    # Rayleigh damping
+    zsponge = FT(1500.0)
+    # TODO: u_relaxation isn't in
+    #rayleigh_sponge = RayleighSponge{FT}(zmax, zsponge, c_sponge, u_relaxation)
+    rayleigh_sponge = RayleighSponge{FT}(zmax, zsponge, c_sponge)
+    # Geostrophic forcing
+    geostrophic_forcing = GeostrophicForcing{FT}(f_coriolis, u_geostrophic, v_geostrophic)
+
+    # Boundary conditions
+    # SGS Filter constants
+    C_smag = FT(0.18)
+    C_drag = FT(0.0011)
+    LHF    = FT(115)
+    SHF    = FT(15)
+    bc = DYCOMS_BC{FT}(C_drag, LHF, SHF)
+
+    config = CLIMA.LES_Configuration("DYCOMS", N, resolution, xmax, ymax, zmax,
+                                     init_dycoms!,
+                                     ref_state=ref_state,
+                                     radiation=radiation,
```

Comment:
I believe we may need to add `ConstantSubsidence{FT}(D_subsidence)` for Dycoms. @smarras79 please confirm.

Reply:
Not trying to slow things down, but a sponge may also be needed. I think we're currently working on updating the sponge in #619.